components/LoggingForm/index.js

```diff
@@ -0,0 +1,86 @@
+import styled from "styled-components";
+import { nanoid } from "nanoid";
+import { useState } from "react";
+
+const StyledForm = styled.form`
+  display: grid;
+  width: 100%;
+  grid-template-columns: repeat(5, 1fr);
+`;
+
+const StyledInput = styled.input`
+  width: 3.5rem;
+`;
+
+export default function LoggingForm({
+  onLog,
+  onSubmit,
+  editMode,
+  editSet,
+  onEditSave,
+}) {
+  const [newData, setNewData] = useState({
+    reps: editMode ? editSet.reps : "",
+    weight: editMode ? editSet.weight : "",
+  });
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const setData = Object.fromEntries(formData);
+
+    if (editMode) {
+      const updatedSet = {
+        ...editSet,
+        ...newData,
+      };
+      onSubmit((prevState) => {
+        const updatedState = [...prevState];
+        const indexOfSet = updatedState.findIndex(
+          (set) => set.id === updatedSet.id
+        );
+        updatedState[indexOfSet] = updatedSet;
+        console.log(updatedSet);
+        return updatedState;
+      });
+      onEditSave((prevState) => !prevState);
+    } else {
+      setData.id = nanoid(5);
+      onSubmit((prevState) => [...prevState, setData]);
+    }
+    onLog((prevState) => !prevState);
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setNewData((prevState) => ({
+      ...prevState,
```

Comment:
Do you really want to add your previous State into the "newData"? I would recommend to log you state and check if the data collected in the state is really needed. If yes maybe you can find a more speaking name than "newData" :)

Reply:
I've just changed the state to this:

```
const [updatedSetValues, setUpdatedSetValues] = useState({
    reps: exercise.reps,
    weight: exercise.weight,
  });
```

The default values specified in the template are now the form's default values. Seems like now this makes more sense but I could be wrong:

```suggestion
    setNewData((prevState) => ({
      ...prevState,
```